frontend/src/pages/warehouse/Warehouses.js

```diff
@@ -0,0 +1,185 @@
+    });
+    const [errorMessage, setErrors] = useState('');
+
+    const handleCheckedChange = (warehouseId) => {
+        const index = checkBoxes.indexOf(warehouseId);
+        if (index > -1) {
+            checkBoxes.splice(index, 1);
+        } else {
+            checkBoxes.push(warehouseId);
+        }
+    };
+
+    const handleCountPerPage = (e) => {
+        e.preventDefault();
+        setPage(preState => ({
+            ...preState,
+            countPerPage: e.target.value
+        }));
+        getWarehouses(`/customers/${props.currentCustomerId}/warehouses?size=${e.target.value}`);
+    };
+
+    const changePage = (e) => {
+        e.preventDefault();
+        let currentPage = e.target.innerHTML - 1;
+        getWarehouses(`/customers/${props.currentCustomerId}/warehouses?page=${currentPage}&size=${page.countPerPage}`);
+    };
+
+    useEffect(() => {
+        getWarehouses(`/customers/${props.currentCustomerId}/warehouses?size=${page.countPerPage}`);
+    }, []);
+
+    function getWarehouses(url) {
+        fetch(url)
+            .then(response => response.json())
+            .then(commits => {
+                setWarehouses(commits.content);
+                setPage({
+                        countPerPage: commits.size,
+                        countPages: commits.totalPages
+                    }
+                );
+            });
+    }
+
+    const closeModalEdit = (e, warehouseDto) => {
+
+        setEditWarehouse(
+            preState => ({
+                ...preState,
+                editShow: false
+            }));
+        if (warehouseDto) {
+            getWarehouses(`/customers/${props.currentCustomerId}/warehouses?size=${page.countPerPage}`);
+        }
+    };
+
+    const closeModalAdd = (e, warehouseDto) => {
+        setLgShow(e);
+        if (warehouseDto) {
+            getWarehouses(`/customers/${props.currentCustomerId}/warehouses?size=${page.countPerPage}`);
+        }
+    };
+
+    function deleteWarehouse() {
+        fetch(`/customers/${props.currentCustomerId}/warehouses/delete-list`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(checkBoxes)
+        })
+            .then(response => {
+                if (response.status !== 204) {
+                    setErrors({
+                        errorMessage: "Warehouse can not be deleted, because it is already used in application"
+                    })
+                } else {
+                    setCheckBox([]);
+                    getWarehouses(`/customers/${props.currentCustomerId}/warehouses?size=${page.countPerPage}`);
+                }
+            });
+    }
+
+    const tableRows = warehouses.map(warehouse => (
+        <tr key={warehouse.id}>
+            <td>{warehouse.identifier}</td>
+            <td>{warehouse.type}</td>
+            <td>{warehouse.addressDto.city}, {warehouse.addressDto.addressLine1},
```

Comment:
We may create a function to return the total address cell content. It will receive a parameter addressDto and return whatever we want.